### PR TITLE
Fix some staticcheck and gosimple issues

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @dougm,

PTAL, I fixed here some **staticcheck** and **gosimple** issues.

### CODE REVIEW

File: simulator/host_datastore_browser.go [hunk 1]
```diff
@@ -128,12 +128,13 @@ func (s *searchDatastore) queryMatch(file os.FileInfo) bool {
 			}
 		case *types.VmDiskFileQuery:
 			if ext == ".vmdk" {
-				if strings.HasSuffix(name, "-flat.vmdk") {
-					// only matches the descriptor, not the backing file(s)
-					return false
-				}
+				// if strings.HasSuffix(name, "-flat.vmdk") {
```
Comment (dougm) [inline simulator/host_datastore_browser.go]: Can we just remove this commented out code?

File: simulator/host_datastore_browser.go [hunk 2]
```diff
@@ -128,12 +128,13 @@ func (s *searchDatastore) queryMatch(file os.FileInfo) bool {
...
+				//	// only matches the descriptor, not the backing file(s)
+				//	return false
+				//}
 				// TODO: check Filter and Details fields
-				return true
+				//return true
```
Comment (dougm) [inline simulator/host_datastore_browser.go]: And remove this too?

File: simulator/host_datastore_browser.go (hunk 1 above)
Comment (mjtrangoni) [reply simulator/host_datastore_browser.go]: sure

File: simulator/host_datastore_browser.go (hunk 2 above)
Comment (mjtrangoni) [reply simulator/host_datastore_browser.go]: sure

**Comment by mjtrangoni** [comment]
@dougm PTAL

**Pull request merged**